maintenance-test-scripts/src/main.rs

```diff
@@ -0,0 +1,334 @@
+use std::str::FromStr;
+use solana_sdk::{
+    commitment_config::CommitmentConfig,
+    hash::{ hash, Hash, },
+    pubkey::{ Pubkey },
+    instruction::{ Instruction },
+    transaction::{ Transaction },
+    signer::{
+        Signer,
+        keypair::{ Keypair, read_keypair_file },
+    },
+    bpf_loader_upgradeable::{
+        self,
+        set_buffer_authority,
+        set_upgrade_authority,
+        UpgradeableLoaderState,
+    },
+};
+
+use solana_client::rpc_client::RpcClient;
+
+use borsh::{BorshDeserialize};
+
+use maintenance::{
+    instruction::{
+        create_maintenance,
+        set_delegate,
+        set_code_hashes,
+        upgrade,
+        set_authority,
+        close_maintenance,
+        get_maintenance_record_address,
+    },
+    state::{
+        MaintenanceRecord,
+    },
+};
+
+const MAINTENANCE_KEY_FILE_PATH: &'static str = "../artifacts/maintenance.keypair";
+const MAINTAIN_AUTHORITY_KEY_FILE_PATH: &'static str = "../artifacts/voter1.keypair";
+const NEW_MAINTAIN_AUTHORITY_KEY_FILE_PATH: &'static str = "../artifacts/voter2.keypair";
+const ADDIN_INITIAL_AUTHORITY_KEY_FILE_PATH: &'static str = "../artifacts/voter3.keypair";
+const ADDIN_KEY_FILE_PATH: &'static str = "../artifacts/addin-fixed-weights.keypair";
+const BUFFER_ADDRESS: &'static str = "8E321qArVoH3WtVA77o7Tu5MHoYqV8St1N1UGMGQhYr3";
+
+fn main() {
```

Comment:
Maybe write tests like https://github.com/neonlabsorg/neon-spl-governance/blob/main/addin-vesting/program/tests/functional.rs instead of this script?

Reply:
Will be updated in another branch.